java/src/main/java/com/google/cloud/dataproc/templates/databases/RedshiftToGCS.java

```diff
@@ -82,5 +81,7 @@ public void runTemplate() {
     spark.stop();
   }
 
-  public void validateInput() {}
+  public void validateInput() {
+    ValidationUtil.validateOrThrow(config);
```

Comment:
Calling ValidationUtil.validateOrThrow(config) from validateInput() causes the unit test to fail since the logic is such that it expects the validation exception to be thrown in of().

Calling validateInput() from of() instead of runTemplate() would resolve this issue.